### ISSUE TITLE
Globally installed ESLint can't find installed config module

### ISSUE THREAD

**Issue by rnickson** [eslint/eslint#10533]
**Tell us about your environment**

* **ESLint Version:** v5.0.1 (`eslint -v`)
* **Node Version:** v10.5.0
* **npm Version:** 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
    "extends": "airbnb-base"
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

Set up ESLint config with:
```bash
m-karas@m-ka:~/wykop-v2$ eslint --init
? How would you like to configure ESLint? Use a popular style guide
? Which style guide do you want to follow? Airbnb
? Do you use React? No
? What format do you want your config file to be in? JSON
Checking peerDependencies of eslint-config-airbnb-base@latest
Installing eslint-config-airbnb-base@latest, eslint-plugin-import@^2.12.0
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN eslint-plugin-import@2.13.0 requires a peer of eslint@2.x - 5.x but none is installed. You must install peerdependencies yourself.
npm WARN eslint-config-airbnb-base@13.0.0 requires a peer of eslint@^4.19.1 but none is installed. You must install peer dependencies yourself.

+ eslint-plugin-import@2.13.0
+ eslint-config-airbnb-base@13.0.0
added 63 packages from 40 contributors in 7.564s
Successfully created .eslintrc.json file in /home/m-karas/wykop-v2
```
And tried this command:

```bash
eslint src/index.js
```

**What did you expect to happen?**
I expected ESLint to lint a file like this:
```bash
m-karas@m-ka:~/wykop-v2$ npx eslint src/index.js

/home/m-karas/wykop-v2/src/index.js
   1:23  error  Unable to resolve path to module 'axios'       import/no-unresolved
   4:1   error  Expected indentation of 2 spaces but found 4   indent
   4:37  error  'options' is defined but never used            no-unused-vars
   5:1   error  Expected indentation of 4 spaces but found 8   indent
   6:1   error  Expected indentation of 6 spaces but found 12  indent
   7:1   error  Expected indentation of 4 spaces but found 8   indent
   8:1   error  Expected indentation of 2 spaces but found 4   indent
  12:1   error  Expected indentation of 2 spaces but found 4   indent
  12:5   error  'request' is not defined                       no-undef

✖ 9 problems (9 errors, 0 warnings)
  6 errors, 0 warnings potentially fixable with the `--fix` option.

(node:23716) [ESLINT_LEGACY_OBJECT_REST_SPREAD] DeprecationWarning: The 'parserOptions.ecmaFeatures.experimentalObjectRestSpread' option is deprecated. Use 'parserOptions.ecmaVersion' instead. (found in "airbnb-base")
```

**What actually happened? Please include the actual, raw output from ESLint.**
```bash
m-karas@m-ka:~/wykop-v2$ eslint src/index.js
Cannot find module 'eslint-config-airbnb-base'
Referenced from: /home/m-karas/wykop-v2/.eslintrc.json
Error: Cannot find module 'eslint-config-airbnb-base'
Referenced from: /home/m-karas/wykop-v2/.eslintrc.json
    at ModuleResolver.resolve (/usr/local/lib/node_modules/eslint/lib/util/module-resolver.js:72:19)
    at resolve (/usr/local/lib/node_modules/eslint/lib/config/config-file.js:484:28)
    at load (/usr/local/lib/node_modules/eslint/lib/config/config-file.js:556:26)
    at configExtends.reduceRight (/usr/local/lib/node_modules/eslint/lib/config/config-file.js:430:36)
    at Array.reduceRight (<anonymous>)
    at applyExtends (/usr/local/lib/node_modules/eslint/lib/config/config-file.js:408:26)
    at loadFromDisk (/usr/local/lib/node_modules/eslint/lib/config/config-file.js:528:22)
    at Object.load (/usr/local/lib/node_modules/eslint/lib/config/config-file.js:564:20)
    at Config.getLocalConfigHierarchy (/usr/local/lib/node_modules/eslint/lib/config.js:227:44)
    at Config.getConfigHierarchy (/usr/local/lib/node_modules/eslint/lib/config.js:179:43)
```

`eslint-config-airbnb-base` is installed in `node_modules` dir, and `npx eslint src/index.js` works as expected. Installing config globally doesn't solve it.

**Comment by platinumazure** [comment]
`eslint --init` installs dependencies locally, so it makes sense that `npx eslint` works but `eslint` (which calls the global installation) doesn't, at least when you only have the local modules.

I'm not sure why installing the dependency globally didn't fix the global ESLint run.

We generally recommend installing ESLint and dependencies locally. It makes for a more reliable experience and it's better for collaboration as well.

**Comment by Kronuz** [comment]
Do eslint plugin modules *must* be siblings of the running eslint? i.e. if we are using `/some/path/node_modules/eslint` module to do some linting, do all used plugins ***must*** exist in`/some/path/node_modules`?

**Comment by ilyavolodin** [comment]
@Kronuz Yes, all dependancies (including plugins) must be siblings to the running instance of ESLint.

**Comment by Kronuz** [comment]
Understood, that's what I had been missing. Thank you!

**Comment by nzakas** [comment]
This question appears resolved, so I'm closing the issue. If you still need help, please send a message to our [mailing list](https://groups.google.com/group/eslint) or [chatroom](https://gitter.im/eslint/eslint). Thanks!
